crates/component-macro/src/bindgen.rs

```diff
@@ -248,12 +248,12 @@ fn select_world(
 
                 // This will ignore the package argument due to the fully
                 // qualified name being used.
-                resolve.select_world(pkgs[0], world)
+                resolve.select_world(pkgs, world)
             }
             None => {
                 let worlds = pkgs
                     .iter()
-                    .filter_map(|p| resolve.select_world(*p, None).ok())
+                    .filter_map(|p| resolve.select_world(pkgs, None).ok())
```

Comment:
This logic looks wrong (not using p for filter_map)

Reply:
Indeed, that's going to be fixed when I merge in the latest changes